src/main/kotlin/com/coder/gateway/views/CoderGatewayRecentWorkspaceConnectionsView.kt

```diff
@@ -206,94 +199,42 @@ class CoderGatewayRecentWorkspaceConnectionsView(private val setContentCallback:
-                                            val link = client.url.withPath("/me/${ws.name}/terminal")
-                                            BrowserUtil.browse(link.toString())
-                                        }
-                                    }
-                                },
-                            )
                         }.topGap(gap)
-                        if (deploymentError == null || showError) {
+
+                        connections.forEach { workspaceProjectIDE ->
+                            val enableLinks = listOf(WorkspaceStatus.STOPPED, WorkspaceStatus.CANCELED, WorkspaceStatus.FAILED, WorkspaceStatus.STARTING, WorkspaceStatus.RUNNING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
+                            val inLoadingState = listOf(WorkspaceStatus.STARTING, WorkspaceStatus.CANCELING, WorkspaceStatus.DELETING, WorkspaceStatus.STOPPING).contains(workspaceWithAgent?.workspace?.latestBuild?.status)
+
                             row {
-                                // There must be a way to make this properly wrap?
-                                label("<html><body style='width:350px;'>" + status.third + "</html>").applyToComponent {
-                                    foreground = status.second
+                                if (inLoadingState) {
+                                    icon(AnimatedIcon.Default())
+                                }
+                                label("<html><body style='width:350px;'>" + status.second + "</html>").applyToComponent {
+                                    foreground = status.first
                                 }
                             }
-                        }
-                        connections.forEach { workspaceProjectIDE ->
+
                             row {
                                 icon(workspaceProjectIDE.ideProduct.icon)
-                                cell(
-                                    ActionLink(workspaceProjectIDE.projectPathDisplay) {
-                                        CoderRemoteConnectionHandle().connect { workspaceProjectIDE }
-                                        GatewayUI.getInstance().reset()
-                                    },
-                                )
+                                if (enableLinks) {
+                                    cell(
+                                        ActionLink(workspaceProjectIDE.projectPathDisplay) {
+                                            withoutNull(deployment?.client, workspaceWithAgent?.workspace) { client, workspace ->
+                                                CoderRemoteConnectionHandle().connect {
+                                                    if (listOf(WorkspaceStatus.STOPPED, WorkspaceStatus.CANCELED, WorkspaceStatus.FAILED).contains(workspace.latestBuild.status)) {
+                                                        client.startWorkspace(workspace)
```

Comment:
I opened https://github.com/coder/jetbrains-coder/issues/465 for future improvement